pyproject.toml

```
@@ -84,6 +85,35 @@ filter_files = true
 known_first_party = "poetry.core"
 known_third_party = ["poetry.core._vendor"]
 
+[tool.mypy]
+python_version = "3.6"
+check_untyped_defs = true
+ignore_missing_imports = true
+warn_redundant_casts = true
+warn_unused_configs = true
+warn_unused_ignores = true
+files = "poetry"
```

to exclude typechecking `_vendor`:

```suggestion
files = "poetry"
exclude = "poetry/core/_vendor"
```